plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/resource/FordiacTypeResource.java

```diff
@@ -115,22 +130,28 @@ protected void doSave(final OutputStream outputStream, final Map<?, ?> options)
 			throw new IOException(
 					MessageFormat.format(Messages.FordiacTypeResource_SaveToUnsupportedURI, uri.toString()));
 		}
-		try {
-			final var fbtFile = ResourcesPlugin.getWorkspace().getRoot()
-					.getFile(new Path(this.uri.toPlatformString(true)));
-			var typeEntryForFile = TypeLibraryManager.INSTANCE.getTypeEntryForFile(fbtFile);
-			if (typeEntryForFile == null) {
-				typeEntryForFile = TypeLibraryManager.INSTANCE.getTypeLibrary(fbtFile.getProject())
-						.createTypeEntry(fbtFile);
+		final AbstractTypeExporter exporter = getTypeExporter(getContents().get(0));
+		if (exporter != null) {
+			try (InputStream inputStream = exporter.getFileContent()) {
+				inputStream.transferTo(outputStream);
 			}
-			typeEntryForFile.setTypeEditable(EcoreUtil.copy((LibraryElement) getContents().get(0)));
-			typeEntryForFile.setLastModificationTimestamp(typeEntryForFile.getFile().getModificationStamp());
-			AbstractTypeExporter.saveType(typeEntryForFile, outputStream);
-		} catch (final Exception e) {
-			throw new IOWrappedException(e);
 		}
 	}
 
+	private static AbstractTypeExporter getTypeExporter(final EObject content) {
+		return switch (content) {
+		case final FunctionFBType funcType -> new FCTExporter(funcType);
+		case final AdapterType adpType -> new AdapterExporter(adpType);
+		case final SubAppType subAppType -> new SubApplicationTypeExporter(subAppType);
+		case final FBType fbType -> new FbtExporter(fbType);
+		case final AnyDerivedType dataType -> new DataTypeExporter(dataType);
+		case final AutomationSystem system -> new SystemExporter(system);
+		case final GlobalConstants globalConstants -> new GlobalConstantsExporter(globalConstants);
+		case final AttributeDeclaration attributeType -> new AttributeTypeExporter(attributeType);
+		default -> null;
```

Comment:
Maybe we should also throw an `IOException` if the content is not supported?

Reply:
I think this is a good idea. Better save then sorry.